rust/core-lib/src/editor.rs

```diff
@@ -913,27 +930,35 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.text.measure::<LinesMetric>() + 1
     }
 
-    pub fn plugin_set_fg_spans(&mut self, start: usize, len: usize, spans: Vec<Span>, rev: usize) {
+    pub fn plugin_add_scopes(&mut self, plugin: &str, scopes: Vec<Vec<String>>) {
+        self.style_scopes.add_scopes(plugin, scopes, &self.doc_ctx);
```

Comment:
Interesting, I hadn't really thought about a plugin wanting more than one layer, but given the limitation on spans overlapping, I do think it makes sense as a future feature.

It's worth noting that using the plugin name as the layer identifier is temporary. Eventually we will assign plugins PIDs; and layers might come from core, as well.